### ISSUE TITLE
MozJPEG lossless transform

### ISSUE THREAD

**Issue by jakearchibald** [GoogleChromeLabs/squoosh#85]
I don't know if this should be V1 or not…

Some JPEG encoders support lossless transforms, such as MozJPEG. We should enable this.

Currently our encoders take image data, so we need to figure out when it's acceptable to pass through the source file directly.

The options UI may need to be aware of when lossless transforms are possible.

**Comment by ZeppLu** [comment]
![image](https://user-images.githubusercontent.com/17007549/61372412-a1536200-a8c9-11e9-9269-361d82e4343f.png)
Maybe I'm wrong, but doesn't a 100% quality indicate lossless transform?

**Comment by jakearchibald** [comment]
It doesn't, it means "run the MozJPEG encoder with a quality level of 100".

**Comment by jakearchibald** [comment]
(that's how it works in Photoshop etc too)

**Comment by surma** [comment]
Yeah, even the setting of 100 will just use a higher resolution quantization table for the DCT transform. That part might be _close_ to lossless, but it’s not guaranteed. Before we get to that stage, tho, the codec will:

- Convert RGB to YUV
- Will subsample U&V to half the original resolution

which are all lossy steps.

**Comment by jakearchibald** [comment]
(I think mozjpeg disables chroma subsampling at qualities > 90)

**Comment by ziemek99** [comment]
> (I think mozjpeg disables chroma subsampling at qualities > 90)

You're right, MozJPEG disables it at qualities >= 90. You can use [this image](https://user-images.githubusercontent.com/15923635/131199794-8691214a-52dc-42bd-af9c-1fecbae7297d.png) to make sure, it'll get **darker** if chroma subsampling is enabled.
